lib/matplotlib/backends/gtk3_compat.py

```diff
@@ -0,0 +1,24 @@
+try:
+    import gi
+except ImportError:
+    try:
+        import pgi as gi
+    except ImportError:
+        raise ImportError("Gtk3 backend requires pygobject to be installed.")
+
+try:
+    gi.require_version("Gtk", "3.0")
+except AttributeError:
+    raise ImportError(
+        "pygobject version too old -- it must have require_version")
+except ValueError:
+    raise ImportError(
+        "Gtk3 backend requires the GObject introspection bindings for Gtk 3 "
+        "to be installed.")
+
+try:
+    pkgs = ['Gtk', 'Gdk', 'GObject', 'GLib']
+    _temp = __import__(gi.__name__+'.repository', globals(), locals(), pkgs, 0)
+    Gtk, Gdk, GObject, GLib = _temp.Gtk, _temp.Gdk, _temp.GObject, _temp.GLib
+except (ImportError, AttributeError):
+    raise ImportError("Gtk3 backend requires pygobject to be installed.")
```

Comment:
Just double checked...
We have:
+ backend_agg
+ backend_cairo
+ backend_gtk3
+ backend_gtk3agg
+ backend_gtk3cairo

and AFAIK we can start ``backend_cairo`` without gtk, so I don't think we should move the cairo stuff into the gtk stuff... if anything it should go into its own file as I do here to share the import logic with the new gtk test I add in this PR thanks to ``pgi``.